api/graph/helpers.go

```diff
@@ -25,3 +28,71 @@ func handleNoRowsOnNullableType[T any, U any](result T, err error, mapper func(T
 
 	return mapper(result), nil
 }
+
+type Cursor[T any] interface {
+	Encode() (string, error)
+	Decode(string) (*Cursor[T], error)
+	Validate(input T) bool
+}
+
+type ListRecipesCursor struct {
+	Id    int
+	Limit int
+}
```

Comment:
Nitpick: We should add JSON struct tags to ensure consistent field names in API responses irrespective of the case-sensitivity and casing.

```suggestion
type ListRecipesCursor struct {
        Id    int `json:"id"`
        Limit int `json:"limit"`
}
```

Reply:
These never get returned by the API, as they are encoded into the cursor, so I don't think this is relevant.